subsys/bluetooth/host/hci_core.c

```diff
@@ -3627,6 +3629,53 @@ int bt_enable(bt_ready_cb_t cb)
 	return 0;
 }
 
+int bt_disable(void)
+{
+	int err;
+
+	if (!bt_dev.drv) {
+		BT_ERR("No HCI driver registered");
+		return -ENODEV;
+	}
+
+	if (!bt_dev.drv->close) {
+		return -ENOTSUP;
+	}
+
+	if (atomic_test_and_set_bit(bt_dev.flags, BT_DEV_DISABLE)) {
+		return -EALREADY;
+	}
+
+	/* Clear BT_DEV_READY before disabling HCI link */
+	atomic_clear_bit(bt_dev.flags, BT_DEV_READY);
+
+	err = bt_dev.drv->close();
+	if (err) {
+		BT_ERR("HCI driver close failed (%d)", err);
+		return err;
+	}
```

Comment:
Won't this leave the stack in an inconsistent and unrecoverable state? You've cleared the `BT_DEV_READY` flag, but not the `BT_DEV_ENABLE` flag. Should the `BT_DEV_READY` be re-enabled in this error branch?

Reply:
Indeed, this is probably safer to set it back. Done